docs/src/main/asciidoc/_developer-guide/ejb3/EJB_on_Kubernetes.adoc

```diff
@@ -0,0 +1,103 @@
+If you scale your application up to 3 replicas you may expect to name of pods like `wildfly-server-0`, `wildfly-server-1`, `wildfly-server-2`.
+
+It's expected a link:https://kubernetes.io/docs/concepts/services-networking/service/#headless-services[headless services]
+to be used along with the `StatefulSet`. With the headless service, there is ensured the DNS hostname for the pod.
+If the WildFly is run on `StatefulSet` with the headless service named `wildfly-server-headless`.
+Then the DNS name of the very first pod will be `wildfly-server-0.wildfly-server-headless`.
+
+If the Wildfly is run as cluster then this use of the hostname `wildfly-server-0.wildfly-server-headless`
+guarantees that the EJB call may reach any WildFly instance connected to the cluster.
+It's a bootstrap connection which is used to initialize the EJB client
+which gathers the structure of the WildFly cluster as the next step.
+
+
+[[ejb-kubernetes-configuration]]
+== EJB configuration for Kubernetes
+
+These are steps you need to process in order to run EJB remote calls.
+Some steps are related to the server configuration, the other ones
+on your application.
+
+. Configure the WildFly cluster on Kubernetes.
+  See the link:High_Availability_Guide{outfilesuffix}#discovery-for-kubernetes[High Availability Guide, section of Kubernetes discovery].
+. All the `socket-binding` must have defined a client mapping for the DNS value mapped by `StatefulSet` headless service.
+  For example in the application which is named `wildfly-server` and considering the `StatefulSet` headless service is named `wildfly-server-headless`
+  then the `http` socket binding needs to be defined in the following way
+
+  ```
+  <socket-binding name="http" port="${jboss.http.port:8080}">
+     <client-mapping destination-address="${jboss.node.name}.wildfly-server-headless"/>
+  </socket-binding>
+  ```
+
+. There is need a small workaround for remote EJB transaction recovery on Kubernetes
+  (the issue could be tracked at link:https://issues.jboss.org/browse/WFCORE-4668[WFCORE-4668]).
+  The WildFly application server has to be configured with property `wildfly.config.url`.
+  The `wildfly.config.url` points to a XML configuration file. If we consider one being placed at `$JBOSS_HOME/standalone/configuration/wildfly-config.xml`
+  then the property is setup as `JAVA_OPTS="$JAVA_OPTS -Dwildfly.config.url=$JBOSS_HOME/standalone/configuration/wildfly-config.xml"`.
+  The `wildfly-config.xml` defines the EJB recovery authentication to be used during transaction recovery for remote EJB calls.
+  The target server has to configure a user that is permitted to receive the EJB remote calls.
+  Such a user is then configured by standard means of link:Admin_Guide{outfilesuffix}#application-realm[security configuration].
+  Let’s say there is configured a user on the target server. It's used the script ``$JBOSS_HOME/bin/add-user.sh` and the user is created under the `ApplicationRealm`.
```

Comment:
"It’s used the script $JBOSS_HOME/bin/add-user.sh" -> "The script $JBOSS_HOME/bin/add-user.sh is used" ?